src/http/base.ts

```diff
@@ -13,6 +13,16 @@ export default class BaseHTTPClient {
         this.config = config;
     }
 
+    protected static errorHandler(reject: any) {
+        return (error: Error | APIError) => {
+            if (error instanceof APIError) {
+                reject(new Error(APIError.mapErrorMessage(error)));
+            } else {
+                reject(error);
+            }
```

Comment:
Something like this is better
```reject(
  (error instanceof APIError)
    ? new Error(APIError.mapErrorMessage(error))
    : reject(error)
);
```

Reply:
that's great! But as discussed with you offline we better leave it for readability